Preserve OwnerReferences when updating Ingresses with a Shared Load Balancer

bittermandel opened the pull request:
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [x] Thanks for contributing!

<!-- Description of change -->

When the IngressController reconciles a Ingress object which is configured to use a Shared LoadBalancer, each field is copied over from a DeepCopy of the original Ingress and the Status.LoadBalancer field is set to the values of the Shared LoadBalancer.

During this process, the OwnerReferences field is discarded. This causes issues when using other operators, such as Cert Manager, which relies on OwnerReferences during its reconcile process.

This fix makes sure the update process copies over the OwnerReferences field as well.

julianwiedmann committed:
bpf: clean up CB_NAT

CB_NAT is no longer used. Instead just define a generic CB_3 placeholder.

Signed-off-by: Julian Wiedmann <jwi@isovalent.com>

sayboras committed:
metrics: Add map pressure metric for auth map

This commit is to make sure that we are having map pressure metric for
auth map to improve observability, which can be useful for adjusting
map size if required.

Relates: https://github.com/cilium/cilium/issues/24617

Signed-off-by: Tam Mach <tam.mach@cilium.io>

julianwiedmann committed:
bpf: encap: clean up usage of __encap_and_redirect_with_nodeid()

Several callers pass src_ip = 0 and vni = NOT_VTEP_DST, and can thus be
switched to the simpler encap_and_redirect_with_nodeid() variant.

Signed-off-by: Julian Wiedmann <jwi@isovalent.com>

bimmlerd committed:
bgp/speaker: use Resource[T] instead of NodeChains

As part of a refactoring effort to remove the CiliumNodeChain, remove
the dependency of the MetalLBSpeaker on it, by directly using the
LocalCiliumNodeResource.

Note that the speaker always discarded events not pertaining to the
local Node, hence no functional change results from the fact that the
CiliumNodeChain proagated events from all cluster CiliumNodes.

As an added benefit, the CiliumNode events now don't stop coming once we
connect to KVStore, since the resource doesn't stop.

Signed-off-by: David Bimmler <david.bimmler@isovalent.com>

bimmlerd committed:
ipam: multipool: use resource for node events

As part of a refactoring effort to remove the CiliumNodeChain, remove
the multipool IPAM dependency on it.

Since IPAM isn't yet part of the hive, we need to pass the resource down
to IPAM through the daemon, in `startIPAM`. Once IPAM code is
modularised, this is no longer necessary.

In order to keep the tests largely unchanged, we reimplement the
fakeNodeK8sAPI testing infrastructure in terms of the resource API. This
leads to code duplication, but a commit later in this series removes the
duplication again, by also converting the other usages to the resource style.

Signed-off-by: David Bimmler <david.bimmler@isovalent.com>

bimmlerd committed:
watchers: remove ciliumnodechain

Previous commits in the series have removed all subscribers to the
CiliumNodeChain by moving them over to use the LocalCiliumNodeResource.
This commit reaps the benefits by deleting all of the now unused code.

Signed-off-by: David Bimmler <david.bimmler@isovalent.com>

joamaki committed:
resource: Add test for repeated deletions

Resource[T] does not correctly handle the events:
  Upsert -> Delete with Done(not-nil) -> Upsert (recreate)
    -> Delete (retry)

The retried delete event carries the old initial version of the
object causing the recreated object to be incorrectly deleted.

Signed-off-by: Jussi Maki <jussi@isovalent.com>

joamaki committed:
resource: Fix double upserts on subscribe and retrying of delete events

Fix double upserts that were caused by store being manipulated without
synchronization with the subscriber queues by processing the deltas
under the resource read-lock and doing the initial key listing for
new subscriber with a write-lock. This way we cannot accidentally
see a key in the store and process it just before the key is queued.

As shown by test case in previous commit, the delete events are retried
with an old incorrect version of the object causing a recreated object
to be deleted.

Fix the deletion retrying by always queueing upserts and deletes by key
and keeping the last known state of objects emitted to the subscriber.
Only emit a delete event if the subscriber has seen its creation and
only use a version of the object that the subscriber has observed.

Fixes: 4101e2c768 ("k8s: Add resource package")
Signed-off-by: Jussi Maki <jussi@isovalent.com>

nathanjsweet committed:
policy: Make MapState an interface

`MapState` has many methods, some of which are public.
In order to keep state to accomplish creating a cohesive
map state that can keep state besides what will be put into
an ebpf map, `MapState` can no longer be a type declaration
for a golang map, but must be a struct that keeps state.

The endpoint package must also be refactored to use `MapState`
as an interface rather than a instantiated type of a map.

Signed-off-by: Nate Sweet <nathanjsweet@pm.me>
Signed-off-by: David Bimmler <david.bimmler@isovalent.com>

meyskens committed:
Setup renovate for SPIRE deployment

This adds a check for the spire-server and spire-agent Docker images for
Renovate.
I also fixes the busybox image to be allowed newer than 1.35.

Signed-off-by: Maartje Eyskens <maartje.eyskens@isovalent.com>

joamaki committed:
resource: Improve documentation on Events()

The semantics around retrying and the Sync event are subtle. Spell out the properties
in a comment for Events().

Signed-off-by: Jussi Maki <jussi@isovalent.com>

chaunceyjiang committed:
cilium ingress support set the number of trusted loadbalancer hops

(as per Envoy config xff_num_trusted_hops)

Fixes: #24292

Signed-off-by: chaunceyjiang <chaunceyjiang@gmail.com>

derailed committed:
Resiliency: Checks endpoints BPF programs remain loaded

We've seen instance where BPF programs are deleted by third party
software (aka Ddog). When this occurrs either agents could fail to
restart or connectivity may be compromised.

This PR attempts to reconciliate BPF program state with their associated
endpoints by ensuring endpoints programs are reloaded when necessary.

- Add a watchdog to ensure endpoints bpf program remain loaded
- Checks for tc qdisk/ingress filter presence
- Add configuration to enable/disable this feature and configure
  watchdog rate
- Add call to terminate controllers upon daemon shutting down

Signed-off-by: Fernand Galiana <fernand.galiana@isovalent.com>

jrajahalme committed:
envoy: Update to cilium-envoy with enhanced L7 LB policy enforcement

Update cilium-envoy image to main that now has enhanced support for L7 LB
policy enforcement.

Signed-off-by: Jarno Rajahalme <jarno@isovalent.com>

jrajahalme committed:
envoy: Fix support for allow-all network policies

Generate the ingress/egress network policy also when l4 filter is
nil. This enables creating allow-all rules when policy is not enforced.

Signed-off-by: Jarno Rajahalme <jarno@isovalent.com>

PlatformLC committed:
endpoint: correct stats for prepareBuild

Previous change relocated CT cleanup blocker, but left prepareBuild
stats calculated after that. Afterwards, some modifications also been
done for the code related to prebuild. So the stats related to prepareBuild
should be handled within runPreCompilationSteps.

Fixes: eaa486d787ef ("endpoint: Wait for CT cleanup to complete before BPF compilation")

Signed-off-by: Li Chun <chun2.li@intel.com>

aanm committed:
.github: re-use common helm values from a single action

A lot of our GH workflows use the same helm values to deploy Cilium.
Thus, it makes sense to re-use one file for that purpose use it as a GH
action composite.

Signed-off-by: André Martins <andre@cilium.io>

yylt committed:
pkg/lbmap: fix incorrect Backend4MapV3Name comment

Signed-off-by: yylt <yang8518296@163.com>

vipul-21 committed:
fix: Remove the latest image tag from docs as latest tag is not published

Signed-off-by: Vipul Singh <vipul21sept@gmail.com>

danehans committed:
BGP CP: API Helper Functions Cleanup

- Moves functions next to types.
- Renames functions to ParseXXX
- Adds godocs.

Signed-off-by: Daneyon Hansen <daneyon.hansen@solo.io>

lmb committed:
pkg/pprof: add CODEOWNER

The package is currently owned by tophat. Let's give it to the last
team that touched it.

Signed-off-by: Lorenz Bauer <lmb@isovalent.com>

nxy7 committed:
Gateway API double slash when stripping path prefix fix

Path prefixes could not be stripped which lead to double slashes.

Fixes: #28270
Signed-off-by: Dawid Danieluk <lolnoxy@gmail.com>

siwiutki committed:
Add hubble_relay_pool_peer_connection_status metric

This change adds a new gauge metric to hubble-relay measuring the connectiion status to all peers. Metric keeps track of number of peers for each possible connectiion status

The current set of metrics is not enough to accurately measure the availability of hubble-relay. They measure the status of gRPC calls, but, for instance, in case all peers are unreachable when GetFlows is called, even though gRPC call will succeed and return "OK" status, the response will come with no flows gathered, rendering it useless. This new metric is introduced to cover such cases.

Signed-off-by: Michal Siwinski <siwy@google.com>

jrajahalme committed:
envoy: Set enforce_policy_on_l7lb on l7 LB listener filters

Turn on ingress policy enforcement on L7 LB. With this cilium-envoy
starts dropping Ingress traffic unless Cilium Agent configures it with a
passing policy via the Ingress endpoint (with the reserved:ingress
identity).

Signed-off-by: Jarno Rajahalme <jarno@isovalent.com>

jrajahalme committed:
k8s: Use reserved:ingress identity also for Gateway API

Tell envoy package to use the IPs allocated for reserved:ingress identity
also when a CEC/CCEC is owned by Gateway API in addition to Cilium
Ingress.

Signed-off-by: Jarno Rajahalme <jarno@isovalent.com>

glrf committed:
Fix data race during Hubble setup in the daemon

This commit fixes a data race where the setup of the hubbleObserver is
being initialized in a new goroutine while the StatusCollector already
tries to check Hubble's status.

We fix the data race by switching to an atomic pointer to access the
`hubbleObserver` in the Daemon struct.

Fixing the data race by simply waiting to start the status collector
until hubble is initialized does not work, as `launchHubble` could take
up to 30s to return while waiting for a TLS certificate and we don't
want to block agent startup for this.

Fixes: #28291

Signed-off-by: Fabian Fischer <fabian.fischer@isovalent.com>

gandro committed:
ci: Avoid using deprecated "tunnel" flag

The tunnel option is deprecated and will be removed in Cilium v1.15.
This commit fixes the remaining uses I have found where the CI
explicitly set the old `tunnel` flag.

Note that the Cilium CLI also still sets the flag some times in our CI,
this is addressed by cilium/cilium-cli#1993.

Signed-off-by: Sebastian Wicki <sebastian@isovalent.com>

squeed committed:
ipcache: aggregate labels from all IPs with local host identity

The `reserved:host` identity is special: the numeric identity is fixed and the
set of labels is mutable. (The datapath requires this.) So, we need to
determine all prefixes that have the `reserved:host` label and capture their
labels. Then, we must aggregate *all* labels from all IPs and insert them as
the `reserved:host` identity labels.

However, the code as written has a race condition whenever the local node has
more than one IP address. This can happen when, for example vxlan or ipv6 is
enabled. The basic sequence is this:

1. Insert IP A as `reserved:host` in to the ipcache. ID 1 now has labels
  `reserved:host`
2. Insert IP A as `reserved:kube-apiserver` in to the ipcache. ID 1 is updated
  with labels `reserved:host, reserved:kube-apsierver`
3. Insert IP B as `reserved:host` in to the ipcache. ID 1 is updated with
  labels `reserved:host`. And now policies that select
  `reserved:kube-apiserver` are broken

Likewise, we need to always update the SelectorCache; we cannot short-circuit
if the ipcache already has that identity. Again, this is needed because the
identity is mutable. So this bug can take another form:

1. Insert IP A as `reserved:host` in to the ipcache. Because IP A is not known
to the ipcache, treat ID 1 as a new identity and update the selector cache
2. Insert IP A as `reserved:kube-apiserver`. Mutate the labels of ID 1. But,
because IP A already has ID 1, short-circuit the update to the selector cache
(if the Source is the same, which it _may_ be).
3. Now the selector cache has incorrect labels for ID 1.

Without this, when there are multiple IPs with the host label, the identity may
flap and the SelectorCache may be missing updates.

Fixes: #28259
Fixes: e0d403adc
Fixes: 308c14225

Signed-off-by: Casey Callendrello <cdc@isovalent.com>

pippolo84 committed:
labels/cidr: Fix slice preallocation size

The number of CIDR labels returned from GetCIDRLabels are equal to the
sum of:

- ones+1 labels coming from the prefixes generated in the loop
- an additional one for the reserved "world" label

To improve performance, we should then allocate a slice of size ones+2.

benchstat analysis confirms this reporting 1 allocation less for each
operation, except for the "0.0.0.0/0" corner case.

name                             old time/op    new time/op    delta
GetCIDRLabels/0.0.0.0/0-8           306ns ±36%     299ns ±32%     ~     (p=0.251 n=20+20)
GetCIDRLabels/10.16.0.0/16-8       4.42µs ± 5%    4.17µs ± 3%   -5.65%  (p=0.000 n=19+19)
GetCIDRLabels/192.0.2.3/32-8       8.43µs ± 5%    8.02µs ± 5%   -4.88%  (p=0.000 n=20+20)
GetCIDRLabels/192.0.2.3/24-8       6.79µs ± 5%    6.49µs ± 5%   -4.41%  (p=0.000 n=20+20)
GetCIDRLabels/192.0.2.0/24-8       6.78µs ± 4%    6.46µs ± 4%   -4.68%  (p=0.000 n=19+20)
GetCIDRLabels/::/0-8                329ns ± 2%     338ns ± 3%   +2.54%  (p=0.000 n=19+20)
GetCIDRLabels/fdff::ff/128-8       53.6µs ± 4%    52.1µs ± 6%   -2.66%  (p=0.000 n=19+17)
GetCIDRLabels/f00d:42::ff/128-8    56.8µs ± 4%    55.1µs ± 2%   -3.04%  (p=0.000 n=20+18)
GetCIDRLabels/f00d:42::ff/96-8     42.4µs ± 3%    40.8µs ± 6%   -3.82%  (p=0.000 n=20+20)

name                             old alloc/op   new alloc/op   delta
GetCIDRLabels/0.0.0.0/0-8            640B ± 0%      656B ± 0%   +2.50%  (p=0.000 n=20+20)
GetCIDRLabels/10.16.0.0/16-8       3.75kB ± 0%    3.17kB ± 0%  -15.37%  (p=0.000 n=18+20)
GetCIDRLabels/192.0.2.3/32-8       8.03kB ± 0%    6.88kB ± 0%  -14.34%  (p=0.000 n=16+20)
GetCIDRLabels/192.0.2.3/24-8       7.21kB ± 0%    6.32kB ± 0%  -12.42%  (p=0.000 n=20+19)
GetCIDRLabels/192.0.2.0/24-8       7.21kB ± 0%    6.32kB ± 0%  -12.42%  (p=0.000 n=19+20)
GetCIDRLabels/::/0-8                 640B ± 0%      656B ± 0%   +2.50%  (p=0.000 n=20+20)
GetCIDRLabels/fdff::ff/128-8       30.8kB ± 0%    25.9kB ± 0%  -15.80%  (p=0.000 n=20+19)
GetCIDRLabels/f00d:42::ff/128-8    33.7kB ± 0%    28.8kB ± 0%  -14.45%  (p=0.000 n=19+20)
GetCIDRLabels/f00d:42::ff/96-8     28.3kB ± 0%    25.1kB ± 0%  -11.31%  (p=0.000 n=20+20)

name                             old allocs/op  new allocs/op  delta
GetCIDRLabels/0.0.0.0/0-8            3.00 ± 0%      3.00 ± 0%     ~     (all equal)
GetCIDRLabels/10.16.0.0/16-8         38.0 ± 0%      37.0 ± 0%   -2.63%  (p=0.000 n=20+20)
GetCIDRLabels/192.0.2.3/32-8         70.0 ± 0%      69.0 ± 0%   -1.43%  (p=0.000 n=20+20)
GetCIDRLabels/192.0.2.3/24-8         54.0 ± 0%      53.0 ± 0%   -1.85%  (p=0.000 n=20+20)
GetCIDRLabels/192.0.2.0/24-8         54.0 ± 0%      53.0 ± 0%   -1.85%  (p=0.000 n=20+20)
GetCIDRLabels/::/0-8                 3.00 ± 0%      3.00 ± 0%     ~     (all equal)
GetCIDRLabels/fdff::ff/128-8          450 ± 0%       449 ± 0%   -0.22%  (p=0.000 n=20+20)
GetCIDRLabels/f00d:42::ff/128-8       450 ± 0%       449 ± 0%   -0.22%  (p=0.000 n=20+20)
GetCIDRLabels/f00d:42::ff/96-8        296 ± 0%       295 ± 0%   -0.34%  (p=0.000 n=20+20)

Signed-off-by: Fabio Falzoi <fabio.falzoi@isovalent.com>

renovate[bot] committed:
chore(deps): update all lvh-images main

Signed-off-by: renovate[bot] <bot@renovateapp.com>

renovate[bot] committed:
fix(deps): update all go dependencies main

Signed-off-by: renovate[bot] <bot@renovateapp.com>

bimmlerd committed:
mapstate: split allows and denies

Now that mapState is a struct, we can split the tracking of allows and
denies. This commit should not introduce functional changes, but
prepares us for an optimization in a later commit.

Signed-off-by: David Bimmler <david.bimmler@isovalent.com>

bimmlerd committed:
mapstate: optimize denyPreferredInsert

Since the denyPreferredInsert logic cleanly splits on whether the to be
inserted entry is a deny entry, we can avoid iterating over some
entries: If we are inserting a deny, we are mostly interested in
iteration over existing allows, and vice versa.

In the case where we have a FQDN policy for a hostname which resolves to
many IPs (for example with a fqdn policy covering Amazon's S3), this
becomes a hot function (without performing a lot of useful work). The S3
workload is expected to be heavily skewed towards allow entries for the
many CIDR identities allocated for S3. When we insert such an entry, we
only need to iterate deny entries, of which there are expected to be
much fewer. Now that the tracking is split, we can employ this fact and,
instead of iterating all entries, iterate only the deny entries.

Signed-off-by: David Bimmler <david.bimmler@isovalent.com>

glrf committed:
Remove dependencies on linux probes for Windows builds

We provide CLI binaries for Windows in `cilium/hubble` and
`cilium/cilium-cli`. These repositories import parts of cilium,
including the `pkg/metrics` package. So we need to make sure that
`pkg/metrics` builds on Windows.

Signed-off-by: Fabian Fischer <fabian.fischer@isovalent.com>

dependabot[bot] committed:
build(deps): bump urllib3 from 2.0.4 to 2.0.6 in /Documentation

Bumps [urllib3](https://github.com/urllib3/urllib3) from 2.0.4 to 2.0.6.
- [Release notes](https://github.com/urllib3/urllib3/releases)
- [Changelog](https://github.com/urllib3/urllib3/blob/main/CHANGES.rst)
- [Commits](https://github.com/urllib3/urllib3/compare/2.0.4...2.0.6)

---
updated-dependencies:
- dependency-name: urllib3
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Commit:
ci: update docs-builder

Signed-off-by: Cilium Imagebot <noreply@cilium.io>

gandro committed:
ipam: Remove unused mock function

This removes a unused mock function. It should have been removed as
part of a previous cleanup commit.

Fixes: ae25fca59c20 ("ipam: Remove cluster-pool-v2beta agent implementation")

Signed-off-by: Sebastian Wicki <sebastian@isovalent.com>

aanm committed:
Makefile: fix 'fast' make targets

These targets failed to select the nodes of a kind cluster. This change
selects the right variables that are used to check for the cluster
nodes.

Fixes: 5f88bbffeb50 ("docs: Add Makefile and documentation for "fast" development targets")
Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
Makefile fix kind-install-cilium-fast target

The kind-install-cilium-fast should install Cilium on all available
kind clusters.

Fixes: 5f88bbffeb50 ("docs: Add Makefile and documentation for "fast" development targets")
Signed-off-by: André Martins <andre@cilium.io>

rbankston committed:
helm: add jobLabel to cilium-{agent,operator} and hubble serviceMonitor

Signed-off-by: Ralph Bankston <ralph.bankston@isovalent.com>

Commit:
docs: Update Kubernetes Gateway-API version to v0.8.1

Synchronize the Cilium Gateway API version support description to align with
the commit d4d7ff4282a2 ("gateway-api: Bump the version to v0.8.1").

Signed-off-by: Haiyue Wang <haiyue.wang@intel.com>

jibi committed:
workflows: cilium-config: parametrize egressgw helm values

Signed-off-by: Gilberto Bertin <jibi@cilium.io>

giorio94 committed:
gha: do not hardcode AWS VPC CNI plugin version in conformance-aws-cni

Currently, we are installing a fixed version of the AWS VPC CNI plugin
version (i.e., v1.11) regardless of the Kubernetes version. Yet, this
means that in certain cases we upgrade it, while in others we downgrade,
it, introducing unnecessary churn.

Let's instead use the default one that gets installed with the given
k8s version. Specifically, for the k8s versions currently tested:

* k8s: v1.23 - AWS VPC CNI: v1.10.4-eksbuild.1
* k8s: v1.24 - AWS VPC CNI: v1.11.4-eksbuild.1
* k8s: v1.25 - AWS VPC CNI: v1.12.2-eksbuild.1
* k8s: v1.26 - AWS VPC CNI: v1.12.5-eksbuild.2
* k8s: v1.27 - AWS VPC CNI: v1.12.6-eksbuild.2

Retrieved through:

for MINOR in $(seq 23 27)
do
    echo -n "k8s: v1.$MINOR - AWS VPC CNI: "
    aws eks describe-addon-versions --addon-name vpc-cni \
      --kubernetes-version 1.$MINOR --output yaml | \
      yq '.addons[].addonVersions[] | select(.compatibilities[].defaultVersion == true) | .addonVersion';
done

Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>

jibi committed:
CODEOWNERS: assign .github/actions to github-sec and ci-structure

similarly to what we are already doing with .github/workflows

Signed-off-by: Gilberto Bertin <jibi@cilium.io>

sayboras committed:
gha: Disable HTTPRouteRequestMultipleMirrors test

Temporarily disable this test, de-flake will be tracked under below
issue.

Relates: https://github.com/cilium/cilium/issues/28374
Signed-off-by: Tam Mach <tam.mach@cilium.io>

julianwiedmann committed:
bpf: egressgw: make ct_status an enum

Allow for a tiny bit more type safety.

Signed-off-by: Julian Wiedmann <jwi@isovalent.com>

jrajahalme committed:
daemon: Add ingress endpoint

Add Cilium Endpoint representing Ingress. It is defined without a veth
interface and no bpf programs or maps are created for it. Ingress
endpoint is needed so that the network policy is computed and configured
to Envoy, so that ingress/egress network policy defined for Ingress can
be enforced.

Cilium Ingress is implemented as L7 LB, which is an Envoy redirect on the
egress packet path. Egress CNP policies are already enforced when
defined. Prior to this commit CNPs defined for reserved:ingress identity
were not computed, however, and all traffic was passed through by Cilium
Ingress was allowed to egress towards the backends. When the backends
receive such packets, they are identified as coming from Cilium Ingress,
so any ingress policies at the backends can not discern the original
source of the traffic.

This commit adds a Cilium endpoint for the reserved:ingress identity,
which makes the Cilium node compute and pass policies whose endpoint
selector selects this identity (e.g., by selecting all entities) to
Envoy, so that they can be enforced. Envoy listener will then enforce not
just the egress policy but also the ingress policy for the original
incoming source security identity.

Signed-off-by: Jarno Rajahalme <jarno@isovalent.com>

raphink committed:
Update docs theme

Signed-off-by: Raphaël Pinson <raphael@isovalent.com>

joestringer committed:
chore(deps): update dependency cilium/cilium-cli to v0.15.9

Signed-off-by: Joe Stringer <joe@cilium.io>

rastislavs committed:
bgpv1: Fix BGP component tests using the same VirtualRouter config

As some of the BGP component tests mutate the VirtualRouter config,
we need to make sure that mutated config is not used by the following tests.
That was the case before this fix, as multiple tests are using the
same baseBGPPolicy. Now each test creates a deep copy of the base config,
which is fine to mutate within the test.
This change also ensures that fixture.config.policy always contains
the most up-to-date policy.

Signed-off-by: Rastislav Szabo <rastislav.szabo@isovalent.com>

meyskens commented on USERS.md:
```diff
@@ -679,3 +679,8 @@ Users (Alphabetically)
       U: CNI/ENI Networking, Network policies, Hubble
       Q: @eugenestarchenko
+
+    * N: Molnett
```
Thanks for this, but could we move this to another PR as the review process here is different. Thanks

bittermandel replied:
Got it 👍🏻

qmonnet committed:
ci: update docs-builder

Manual update for docs-builder tag, because the related PR's branch was
not opened in the Cilium repository (but from a fork).

Signed-off-by: Quentin Monnet <quentin@isovalent.com>

aanm committed:
.github/actions/helm-default: use the derived SHA as image tag

This commit ensures that the correct image tag, derived from the GitHub
SHA, is used in the Helm chart when workflows are triggered by events
other than 'workflow_dispatch' or 'pull_request'

Fixes: 4498ec908b02 (".github: re-use common helm values from a single action")

Reported-by: Marco Iorio <marco.iorio@isovalent.com
Signed-off-by: André Martins <andre@cilium.io>

ti-mo committed:
bpf: don't sign-extend IPv6 address components

Unqualified integers are treated as `int` in C, but the behaviour when doing
binary operations on them against unsigned integers is surprising, to say the
least.

Consider the following scenario, before this patch:

```
 #define DEFINE_IPV6(NAME, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16) \
 DEFINE_U64_I(NAME, 1) = bpf_cpu_to_be64( \
                       (__u64)(a1) << 56 | (__u64)(a2) << 48 | (__u64)(a3) << 40 | \
                       (__u64)(a4) << 32 | (a5) << 24 | (a6) << 16 | (a7) << 8 | (a8)) .. <elip>

 DEFINE_IPV6(ROUTER_IP, 0x0, 0x0, 0x0, 0x0, 0x80, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x80, 0x0, 0x0, 0x0);
```

Both a5 and a13 are set to a byte with a most-significant bit of 1. This
results in the following output in the .data section during compilation:

```
0000000000000038 <ROUTER_IP_1>:
       7:       ff ff ff ff 80 00 00 00 <unknown>
0000000000000040 <ROUTER_IP_2>:
       8:       ff ff ff ff 80 00 00 00 <unknown>
```

Since the `<< 24` pushed the 1 bit to now be the most-significant bit of an int,
equivalent to an int32 on most 64-bit architectures, the subsequent OR operation
against a __u64 causes the value to become sign-extended, setting the upper 32
bits of the resulting (unsigned..) integer to 1. Ouch.

This commit treats each member as a u64, and additionally truncates each
macro argument to u8 to prevent any of the arguments overlapping as a
defensive measure.

Signed-off-by: Timo Beckers <timo@isovalent.com>

tklauser committed:
Remove deprecated policy_import_errors_total metric

The policy_import_errors_total metric was deprecated in Cilium 1.14 in
favor of the policy_change_total metric, see commit 7ab330310ed8
("Expand agent metric Policy Import Errors to count all policy changes")
for details.

Signed-off-by: Tobias Klauser <tobias@cilium.io>

giorio94 committed:
gha: explicit branch and trigger in ariane-scheduled workflow

Let's use explicit bash variables to construct the trigger phrase,
instead of directly using the matrix value.

Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>

giorio94 committed:
.github/actions/helm-default: fix kvstoremesh image

The kvstoremesh image repository was incorrectly set to
`clustermesh-apiserver-ci`. Let's fix it.

Fixes: 4498ec908b02 (".github: re-use common helm values from a single action")
Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>

giorio94 committed:
gha: restore kvstoremesh.enabled setting in conformance-clustermesh

This setting has been incorrectly dropped in a previous commit, let's
restore it, so that we keep testing kvstoremesh.

Fixes: 4498ec908b02 (".github: re-use common helm values from a single action")
Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>

bittermandel committed:
fix: preserve ownerreferences when updating ingress

Signed-off-by: Jonathan Grahl <jonathan@keyholders.io>

bittermandel committed:
Merge branch 'main' into patch-1
